### ISSUE TITLE
Error for the VM(64/32)(x/y/z) memory operands in the scatter/gather instructions on X86

### ISSUE THREAD

**Issue by chanhunan** [lifting-bits/remill#540]
When the following instruction, i.e. AVX-512vl: vpscatterqq xmmword ptr [ rsp+xmm0*1]{k1}, xmm1, executes CHECK function, the following questions are reported.
Error:

**[Lifter.cpp:610] Check failed word_size >= arch_addr.index_reg.size
Memory index register XMM0 for instruction at 5457fe 0x1ffa108 is wider than the machine word size.
*** Check failure stack trace: ***
	@	0x7f4d02907bc	google::LogMessageFatal::~LogMessageFatal()
	@	0x553db3	remill::InstructionLifter::LifterAddressOperand()**

The semantic function of the instruction ‘vpscatterqq’ is completed according to the pseudo code: [https://www.felixcloutier.com/x86/vpscatterdd:vpscatterdq:vpscatterqd:vpscatterqq](vpscatterqq)

The code compilation can pass, but the test unit reports the above error.

Meanwhile, I've tentatively modified the code of [Lifter.cpp:610] ‘remill::InstructionLifter::LifterAddressOperand()’: i.e.

CHECK(word_size >= arch_addr.index_reg.size)
      << "\nMemory base register " << arch_addr.base_reg.name << arch_addr.base_reg.size
      << " for instruction at " << std::hex << inst.pc
      << " is wider than the machine word size." << word_size
      << "\nMemory index register " << arch_addr.index_reg.name << arch_addr.index_reg.size
      << " for instruction at " << std::hex << inst.pc
      << " is wider than the machine word size." << word_size;

After I executed the project, I found a puzzling problem: 

[Lifter.cpp:603] CHECK(word_size >= arch_addr.base_reg.size) can pass, 
but [Lifter.cpp:610] CHECK(word_size >= arch_addr.index_reg.size) can not.

The output of the question is as follows:

[Lifter.cpp:610] Check failed word_size >= arch_addr.index_reg.size
Memory base register RSP64 for instruction at 5457fe 0x1ffa108 is wider than the machine word size. 40 
Memory index register XMM080 for instruction at 5457fe 0x1ffa108 is wider than the machine word size. 40

How should I modify to ensure that the test unit passes?

**Comment by pgoodman** [comment]
I don't think we implement the semantics for this instruction, although this check failure shouldn't happen in the first place.

**Comment by chanhunan** [comment]
The following is the semantic function of the instruction ‘vpscatterqq’ under XMM: 

	template <typename D>
    DEF_SEM(VPSCATTERQQ_SIMD128, D dst, R8W k, I8 k1, VV128 src) {
        auto dst_vec = UClearV64(UReadV64(dst)); 
        uint8_t k1_vec = Read(k1); 
        uint8_t k_vec= Read(k); 
        auto src_vec = UReadV64(src);  
        if(k1_vec &0x1){
            dst_vec=UInsertV64(dst_vec,0,UExtractV64(src_vec,0));
			k1_vec =k1_vec & 0xFE;
        }
        if((k1_vec &0x2)==0x2){
            dst_vec=UInsertV64(dst_vec,1,UExtractV64(src_vec,1));
			k1_vec =k1_vec & 0xFD;
        }
        k_vec= k1_vec &0x03; 
        UWriteV64(dst, dst_vec);
        Write(k,k_vec);
        return memory;
    }
#if HAS_FEATURE_AVX512
DEF_ISEL(VPSCATTERQQ_MEMu64_MASKmskw_XMMu64_AVX512_VL128) =VPSCATTERQQ_SIMD128<MV128W>; 
#endif

**Comment by pgoodman** [comment]
Oh what do you know; I suppose we do have it. @kumarak, can you look into this?

**Comment by pgoodman** [comment]
@chanhunan can you give us the hex encoding of the bytes of the instruction to help us reproduce this issue?

**Comment by chanhunan** [comment]
vpscatterqq     %xmm1, (%rsp,%xmm0) {%k1} # encoding: [0x62,0xf2,0xfd,0x09,0xa1,0x0c,0x04]
